tests/testthat/test-test-server.R

```diff
@@ -651,3 +651,79 @@ test_that("session flush handlers work", {
 
   })
 })
+
+test_that("module return value captured", {
+  module_implicit_return <- function(id) {
+    moduleServer(id, function(input, output, session) {
+      123
+    })
+  }
+
+  testServer(module_implicit_return, {
+    expect_equal(session$returned, 123)
+  })
+
+  module_early_returns <- function(id, n) {
+    retval <<- NULL
```

Comment:
This should be `<-`; otherwise it will assign `retval` in the global environment.